### ISSUE TITLE
False positive when iterating over NamedTemporaryFile

### ISSUE THREAD

**Issue by VFLashM** [PyCQA/pylint#1579]
### Steps to reproduce

```python
import tempfile
tfile = tempfile.NamedTemporaryFile()
for line in tfile:
    print line
```

### Current behavior

```
E:  3,12: Non-iterable value tfile is used in an iterating context (not-an-iterable)
```

### Expected behavior

No error

### pylint --version output

```
pylint 1.7.2, 
astroid 1.5.3
Python 2.7.12 (default, Sep 20 2016, 14:05:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce the problem!
